### PR TITLE
VideoPlayer: some codecs have no extradata

### PR DESCRIPTION
**Pull request by Rechi**
## Motivation and Context
VP8 and VP9 videos were not working
@FernetMenta I've picked the patch from your repo and added the VP8 and VP9.

## How Has This Been Tested?
played VP8 & VP9 video

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp [hunk 1]
```diff
@@ -125,7 +125,14 @@ bool CVideoPlayerVideo::OpenStream(CDVDStreamInfo hint)
   if (hint.flags & AV_DISPOSITION_ATTACHED_PIC)
     return false;
   if (hint.extrasize == 0)
-    return false;
+  {
+    // codecs with no extradata
+    if (hint.codec != AV_CODEC_ID_MJPEG &&
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: @peak3d what codecs do require extradata? maybe we should change this to a positive list

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: @Rechi could you reduce extradata check to those codecs only:
AV_CODEC_ID_MPEG1VIDEO
AV_CODEC_ID_MPEG2VIDEO
AV_CODEC_ID_MPEG2VIDEO_XVMC
AV_CODEC_ID_H264
AV_CODEC_ID_HEVC
AV_CODEC_ID_MPEG4
AV_CODEC_ID_WMV3
AV_CODEC_ID_VC1

others are ok without extradata

**Comment by Rechi** [comment]
@FernetMenta changed like you requested

**Comment by FernetMenta** [comment]
looks good, thanks

**Comment by Rechi** [comment]
jenkins build this please

**Pull request merged**